File: backend/spring-intern/src/main/java/com/example/intern_dashboard/service/InternService.java
package com.example.intern_dashboard.service;

import com.example.intern_dashboard.dto.request.InternRequest;
import com.example.intern_dashboard.dto.response.ApiResponse;
import com.example.intern_dashboard.dto.response.InternResponse;

import java.util.List;

public interface InternService {

    ApiResponse createIntern(InternRequest request);

    List<InternResponse> getAllInterns();

    InternResponse getInternById(int id);

    ApiResponse updateIntern(int id, InternRequest request);

    ApiResponse deleteIntern(int id);

    List<InternResponse> searchInterns(String name);

    InternResponse updateIntern(Integer id, InternRequest dto);
}

File: frontend/internhub-frontend/src/services/internService.js
import api from "../utils/api";

export const getAllInterns = async () => {
  const response = await api.get("/api/interns");

  return response.data;
};

export const createIntern = async (internData) => {
  const response = await api.post(
    "/api/interns",
    internData
  );

  return response.data;
};

export const deleteIntern = async (id) => {
  const response = await api.delete(
    `/api/interns/${id}`
  );

  return response.data;
};

export const searchInterns = async (name) => {
  const response = await api.get(
    `/api/interns/search?name=${name}`
  );

  return response.data;
};

export const updateIntern = async (
  id,
  internData
) => {

  const response = await api.put(
    `/api/interns/${id}`,
    internData
  );

  return response.data;

};